index.js

```diff
@@ -92,26 +124,27 @@ function run() {
   cli.prepare({
     cwd: opts.cwd,
     configPath: opts.gulpfile,
-    require: opts.require,
+    preload: opts.preload,
     completion: opts.completion,
-  }, function(env) {
-    var cfgLoadOrder = ['home', 'cwd'];
-    var cfg = loadConfigFiles(env.configFiles['.gulp'], cfgLoadOrder);
-    opts = mergeConfigToCliFlags(opts, cfg);
-    env = mergeConfigToEnvFlags(env, cfg, opts);
-    env.configProps = cfg;
-
-    // Set up event listeners for logging again after configuring.
-    toConsole(log, opts);
-
-    cli.execute(env, env.nodeFlags, handleArguments);
-  });
+  }, onPrepare);
 }
 
 module.exports = run;
 
+function onPrepare(env) {
+  var cfg = mergeProjectAndUserHomeConfigs(env);
+  opts = mergeConfigToCliFlags(opts, cfg);
+  env = mergeConfigToEnvFlags(env, cfg, opts);
```

Comment:
`env` needs to be overridden here, because `env.preload`, `env.configPath` and `env.configBase` can be overridden by `flags.preload`, `flags.gulpfile` in config files, and these properties are used in `Liftoff#execute`.